internal/factory/container/container.go

```diff
@@ -759,3 +762,13 @@ func getOCICapabilitiesList() []string {
 	}
 	return caps
 }
+
+// IsInCRIMounts checks if a mount path is in the list of mounts
+func (c *container) IsInCRIMounts(dst string) bool {
```

Comment:
TBH I dislike this function. I think it's a side effect of poor organization. I think we should refactor all of the mount code so we don't have to constantly loop through the mounts like a bunch of times. Focusing on that would reduce the complexity a lot. WDYT?

Reply:
Yeah, that's a good idea.